depends/packages.md

```diff
@@ -5,6 +5,9 @@ The package "mylib" will be used here as an example
 
 General tips:
 - mylib_foo is written as $(package)_foo in order to make recipes more similar.
+- Secondary dependency packages relative to the bitcoin binaries/libraries (i.e.
```

Comment:
What is the rationale for this? Does this improve build times, or compatibility of some kind?

Reply:
It improves general build reliability.

For example, when linking an executable against a shared library `libprimary` that has its own shared dependency `libsecondary`, we may need to specify the path to `libsecondary` on the link command using the `-rpath/-rpath-link` options, it is not sufficient to just say `libprimary`.

For us, it's much easier to just link a static `libsecondary` into a shared `libprimary`. Especially because in our case, we are linking against a dummy `libprimary` anyway that we'll throw away. We don't care if the end-user has a static or dynamic `libseconday`, that's not our concern. With a static `libseconday`, when we need to link `libprimary` into our executable, there's no dependency chain to worry about, `libprimary` has all the symbols.

Perhaps @theuni can add more.

Reply:
This might be helpful to add as a comment.